### PR TITLE
CI: adding datacite validation 

### PR DESCRIPTION
**Pull request by djarecka**
added `publish=True` to one test to enable datacite validation. The doi will be published on https://doi.test.datacite.org/ as Findable, so will not be able to remove it!

I'm creating `doi` with `id = 000002` and random `version = 0.0.{random.randrange(0, 9999)}`

### CODE REVIEW

File: dandischema/tests/test_datacite.py [hunk 1]
```diff
@@ -53,9 +53,9 @@ def schema() -> Any:
 
 @pytest.fixture(scope="function")
 def metadata_basic() -> Dict[str, Any]:
-    dandi_id_noprefix = f"000{random.randrange(100, 999)}"
+    dandi_id_noprefix = "000002"
     dandi_id = f"DANDI:{dandi_id_noprefix}"
-    version = "0.0.0"
+    version = f"0.0.{random.randrange(0, 9999)}"
```
Comment (satra) [inline dandischema/tests/test_datacite.py:58]: instead of random, perhaps use the current date+time in some form.

**Comment by satra** [comment]
i think it would be good to limit the publishing test to only the platform that dandiarchive uses, so we don't have to test it for all options and limit the number of published dois on the test site.

**Comment by djarecka** [comment]
@satra - you mean just to use for one of the linux containers we have, not adding a new one that is the same as dandiarchive uses, right?

**Comment by satra** [comment]
closing this since #167 was merged.

**Pull request closed without merging**